packages/server/src/adapters/content-handlers/selectContentHandlerOrUnsupportedMediaType.ts

```diff
@@ -1,20 +1,35 @@
 import { TRPCError } from '../../@trpc/server';
-import type { BaseContentTypeHandler } from '../../@trpc/server/http';
+import type {
+  BaseContentTypeHandler,
+  HTTPHeaders,
+} from '../../@trpc/server/http';
+
+interface MinimalHandlerOpts {
+  req: {
+    headers: HTTPHeaders | Headers;
+  };
+}
 
 export function selectContentHandlerOrUnsupportedMediaType<
-  THandlerOpts,
+  THandlerOpts extends MinimalHandlerOpts,
   THandler extends BaseContentTypeHandler<THandlerOpts>,
 >(handlers: THandler[], opts: THandlerOpts) {
-  const handler = handlers.find((handler) => handler.isMatch(opts));
+  const contentType = new Headers(opts.req.headers as HeadersInit).get(
```

Comment:
this cast is fine, idk why it complains that you can't pass a string[] in ther...

![CleanShot 2024-04-21 at 11 02 14@2x](https://github.com/trpc/trpc/assets/51714798/2f16a832-aee6-4f47-a8f5-c3f13dd196ab)